### PR TITLE
[inductor][cpp] unified the vectorized conversion with `at::vec::convert` for all data types

### PR DESCRIPTION
**Pull request by jgong5**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #122961
* #122878
* #122869
* __->__ #119979

This PR unified the vectorized conversion with `at::vec::convert` for all vectorized data types. The intrinsics implementations are implemented as a specialization and moved to their own arch-specific files. The vectorized conversion logic in cpp Inductor is simplified.

cc @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by malfet** [comment]
@jgong5 can you please add tests for those new functions to `vec_test_all_types`, because as of right now there are no way to test AVX2 or default fallbacks by using inductor, and also, it's better to have a unit tests, rather than an integration one. For example, [out-of-bounds access](https://github.com/pytorch/pytorch/pull/122511) error in default `cvt_int64_to_int32` was exposed only when running `test_div2_cpu` inductor test, but it would be much nicer if the same were tested by vec_test_all

**Comment by jgong5** [comment]
> @jgong5 can you please add tests for those new functions to `vec_test_all_types`, because as of right now there are no way to test AVX2 or default fallbacks by using inductor, and also, it's better to have a unit tests, rather than an integration one. For example, [out-of-bounds access](https://github.com/pytorch/pytorch/pull/122511) error in default `cvt_int64_to_int32` was exposed only when running `test_div2_cpu` inductor test, but it would be much nicer if the same were tested by vec_test_all

Thanks. I add the test to https://github.com/pytorch/pytorch/pull/122869 in the same stack. Please review. I didn't add it here to avoid unnecessary conflict since there seems to be other PR extending `vec_test_all_types`. My local test passed. I will add test on `VecMask` in another PR later.

**Comment by jgong5** [comment]
@jansel @malfet The PR is ready. Could you please take a look? Thanks.

**Review by jansel — APPROVED** [review]

File: aten/src/ATen/cpu/vec/vec256/vec256_convert.h [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+template <typename dst_t>
+struct VecConvert<
+    dst_t,
+    1,
+    int64_t,
+    2,
+    typename std::enable_if<
+        std::is_same_v<dst_t, int8_t> ||
+        std::is_same_v<dst_t, uint8_t>>::type> {
```
Comment (malfet) [inline aten/src/ATen/cpu/vec/vec256/vec256_convert.h:137]: Nit
```suggestion
    typename std::enable_if_t<
        std::is_same_v<dst_t, int8_t> ||
        std::is_same_v<dst_t, uint8_t>>> {
```

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**